ucrop/src/main/java/com/yalantis/ucrop/task/BitmapLoadTask.java

```diff
@@ -110,7 +110,23 @@ protected BitmapWorkerResult doInBackground(Void... params) {
         options.inJustDecodeBounds = true;
         BitmapFactory.decodeFileDescriptor(fileDescriptor, null, options);
         if (options.outWidth == -1 || options.outHeight == -1) {
-            return new BitmapWorkerResult(new IllegalArgumentException("Bounds for bitmap could not be retrieved from the Uri: [" + mInputUri + "]"));
+            //return new BitmapWorkerResult(new IllegalArgumentException("Bounds for bitmap could not be retrieved from the Uri: [" + mInputUri + "]"));
+			
+			try {
+                ExifInterface exifInterface = new ExifInterface(mInputUri.getPath());
+                int height = exifInterface.getAttributeInt(ExifInterface.TAG_IMAGE_LENGTH, ExifInterface.ORIENTATION_NORMAL);//获取图片的高度
+                int width = exifInterface.getAttributeInt(ExifInterface.TAG_IMAGE_WIDTH, ExifInterface.ORIENTATION_NORMAL);//获取图片的宽度
+                Log.d(TAG, "exif height: " + height);
+                Log.d(TAG, "exif width: " + width);
+                options.outWidth = width;
+                options.outHeight = height;
+
+                mRequiredWidth=width/3;
```

Comment:
I mean why do you need to set mRequiredHeight = height / 3. What does the "3" means?